net/netdev/netdev_txnotify.c

```diff
@@ -57,17 +57,9 @@
 #ifdef CONFIG_NET_IPv4
 void netdev_ipv4_txnotify(in_addr_t lipaddr, in_addr_t ripaddr)
 {
-  FAR struct net_driver_s *dev;
-
   /* Find the device driver that serves the subnet of the remote address */
 
-  dev = netdev_findby_ripv4addr(lipaddr, ripaddr);
-  if (dev && dev->d_txavail)
-    {
-      /* Notify the device driver that new TX data is available. */
-
-      dev->d_txavail(dev);
-    }
+  netdev_txnotify_dev(netdev_findby_ripv4addr(lipaddr, ripaddr));
```

Comment:
Good! I was worried about netdev_findby_ripv4addr() could return NULL dev, but I confirmed that netdev_txnotify_dev() is checking it before proceeding.

Reply:
Yes, so we can reuse the checking code from netdev_txnotify_dev() and remove the redundant check